### ISSUE TITLE
Why does the main wallet balance differ from the on-chain balance shown on the Create Channel Pay form?

### ISSUE THREAD

**Issue by bolatovumar** [LN-Zap/zap-desktop#2651]
I couldn't send a payment or open a new channel even though the main dashboard was showing that I had balance but when I went to payment/open channel screens my balance was shown as zero.

NOTE: I was using the "next" branch on Testnet. I enabled Testnet by changing `network` property in `default.js` to `'testnet'`.

## Expected Behavior

Balance should be displayed consistently everywhere.

## Actual Behavior

Balance is shown correctly on the main screen but shown as zero on payment/open channel screens.

## Steps to Reproduce

1. Ran `yarn && yarn dev` on `next` branch
2. Requested testnet coins from a faucet
3. Once I received my coins I opened a channel with a node with this id: `0260d9119979caedc570ada883ff614c6efb93f7f7382e25d73ecbeba0b62df2d7`
4. While waiting for the funding transaction to confirm I attempted to open another channel with nodes that Zap suggested
5. Couldn't open any channels because my balance was showing as zero on the "open channel" screen. Same was happening on the "pay" screen
6. Restarted Zap and balance was showing correctly after this restart and now my balance started showing correctly and I could open another channel
7. Once I opened this new channel I encountered the same issue again (balance showing as zero again)
8. I restarted Zap again thinking it would fix the issue like it did last time but it didn't. I attempted this twice with no luck. I don't see any errors in the console.

## Screenshots

Main screen with correct balance:
![Screen Shot 2019-07-30 at 5 11 38 PM](https://user-images.githubusercontent.com/1934678/62139603-51759180-b2f2-11e9-83df-ed55b936b789.png)

Pending channel creation screen:
![Screen Shot 2019-07-30 at 5 11 30 PM](https://user-images.githubusercontent.com/1934678/62139685-71a55080-b2f2-11e9-99f3-84a5b42a7557.png)

"Open a channel" screen with incorrect balance:
![Screen Shot 2019-07-30 at 5 12 06 PM](https://user-images.githubusercontent.com/1934678/62139781-9699c380-b2f2-11e9-9d89-a1aa2b06a2df.png)

"Open a channel" screen with correct balance after Zap restart:
![Screen Shot 2019-07-30 at 5 20 12 PM](https://user-images.githubusercontent.com/1934678/62139949-ce087000-b2f2-11e9-83a8-f98dd20483a7.png)

Terminal output (nothing special):
![Screen Shot 2019-07-30 at 5 43 49 PM](https://user-images.githubusercontent.com/1934678/62140064-0019d200-b2f3-11e9-88e7-080c65ef63f1.png)

Browser console also doesn't have anything special in it except for an unrelated warning:
![Screen Shot 2019-07-30 at 5 46 26 PM](https://user-images.githubusercontent.com/1934678/62140156-27709f00-b2f3-11e9-9d34-0381dc12d40b.png)

## Your Environment

- Zap version: 0.6.0-alpha
- Operating System and version: macOS Mojave v10.14.5

**Comment by bolatovumar** [comment]
Now I received a 1,000,000 satoshis and the "Open a Channel" and "Pay" screen show wrong on-chain balance again but it shows 1,000,000 satoshis instead of 0 as it did previously. Looks like the old balance (0.03 tbtc) is being ignored and only the newly received 0.01 tbtc is being counted.

Notice newly received 1,000,000 satoshis:
![Screen Shot 2019-07-30 at 6 02 26 PM](https://user-images.githubusercontent.com/1934678/62141243-e5e0f380-b2f4-11e9-9559-e93f198d9ca7.png)

See wrong on-chain balances on "Open a channel" and "Pay" screens:
![Screen Shot 2019-07-30 at 6 02 51 PM](https://user-images.githubusercontent.com/1934678/62141356-1c1e7300-b2f5-11e9-97aa-093d7b42389f.png)
![Screen Shot 2019-07-30 at 6 05 39 PM](https://user-images.githubusercontent.com/1934678/62141362-1de83680-b2f5-11e9-8dec-d2f171df4fcc.png)

**Comment by mrfelton** [comment]
@bolatovumar note that the send from only shows you your confirmed balance since this is the only balance that you are able to send from. The main wallet screen shows your total balance including any unconfirmed transactions.

**Comment by bolatovumar** [comment]
@mrfelton both on-chain transactions were confirmed though. You will also see in my second comment that a later transaction balance is shown in the "pay" screen (1mil sats) while the prior transaction balance is not. I will try to investigate further.

**Comment by bolatovumar** [comment]
Looks like the issue is in [lightning.methods.js on line 71](https://github.com/LN-Zap/zap-desktop/blob/next/services/grpc/lightning.methods.js#L71). As far as I understand `this.walletBalance()` makes a call to lnd? It seems like it returns wrong confirmed balance right after you open a channel on testnet.

I recorded a video of this issue:
[https://youtu.be/tEPXD__FjF0](https://youtu.be/tEPXD__FjF0)

Let me know if I just misunderstand something.

**Comment by mrfelton** [comment]
@bolatovumar I suspect this is due to the fact that you have unconfirmed transactions. Hear me out:

In your video, when you went to open the first channel you actually already had a pending channel open operation in progress (the one to `ryomo-6`). After this you had 2 channels in a pending open state. The total amount of UTXOs used in the inputs to these transactions is unspendable until the channel opening transactions are confirmed.

When you open a channel you are making a bitcoin transaction. This transaction has one or more inputs and multiple outputs. One of the outputs goes to the channel multi-sig address amd the other is to a change address in your own wallet. However, until the transaction is confirmed these change coins are unspendable. This means that the total amount of your channel opening transaction inputs is unspendable.

For example, lets say that you had the following UTXOs in our wallet:

A) 1,000,000
B) 1,000,000
C) 300,000

If you initiate a channel opening for 1,100,000 the wallet might select A and C as inputs to the transaction for a total of 1,300,000. That transaction will have 2 outputs - 1,100,000 for the channel multi-sig and 200,000 back to your wallet. The 200,000 that is due to go back to you as change is unspendable until the transaction is confirmed and therefore will not show up in the balances on the send or channel open forms (but will be included in the main wallet balance)

**Comment by bolatovumar** [comment]
@mrfelton right, I see, I didn't think this through. As far as I can tell there is no bug in Zap code around this and it's unlikely to be an LND issue.

Most likely what happened in my case was:
1. I had two UTXOs: 3mil sats and 1mil sats.
2. I opened a channel which used one of the UTXOs (the 3mil one).
3. I attempted to open a new channel while the previous transaction hasn't been confirmed yet. My 3mil UTXO was "locked" at this point and not available for opening a new channel.
4. I was seeing 1mil as my balance because that's the only UTXO available for opening another channel.

Sounds about right?

All of the above makes me wonder if there is a UX improvement that can be made here. 

Perhaps the "pending transactions amount" could be shown alongside the "on-chain balance" amount. 

Alternatively the wording could be adjusted in the footer of the "Open a Channel" screen. Currently it says:

> Your current on-chain balance: 2,553,363 tsatoshis

I wonder if it would make sense to update the wording to something like
> On-chain balance available for opening a channel: 2,553,363 tsatoshis

A similar UX improvement could be made in the "Send Bitcoin" screen.
